src/polyfills/fetch.js

```diff
@@ -0,0 +1,321 @@
+      this.drainText_().then(parseJson));
+  }
+
+  /**
+   * Drains the response and returns a promise that resolves with the response
+   * ArrayBuffer.
+   * @return {!Promise<!ArrayBuffer>}
+   */
+  arrayBuffer() {
+    return /** @type {!Promise<!ArrayBuffer>} */ (
+      this.drainText_().then(utf8Encode));
+  }
+}
+
+/**
+ * Normalized method name by uppercasing.
+ * @param {string|undefined} method
+ * @return {string}
+ * @private
+ */
+function normalizeMethod(method) {
+  if (method === undefined) {
+    return 'GET';
+  }
+  method = method.toUpperCase();
+  dev().assert(
+      allowedMethods.includes(method),
+      'Only one of %s is currently allowed. Got %s',
+      allowedMethods.join(', '),
+      method
+  );
+  return method;
+}
+
+
+/**
+ * Provides access to the response headers as defined in the Fetch API.
+ * @private Visible for testing.
+ */
+class FetchResponseHeaders {
+  /**
+   * @param {!XMLHttpRequest|!XMLHttpRequestDef} xhr
+   */
+  constructor(xhr) {
+    /** @private @const {!XMLHttpRequest|!XMLHttpRequestDef} */
+    this.xhr_ = xhr;
+  }
+
+  /**
+   * @param {string} name
+   * @return {string}
+   */
+  get(name) {
+    return this.xhr_.getResponseHeader(name);
+  }
+
+  /**
+   * @param {string} name
+   * @return {boolean}
+   */
+  has(name) {
+    return this.xhr_.getResponseHeader(name) != null;
+  }
+}
+
+export class Response extends FetchResponse {
+  /**
+   * Returns instance of Response.
+   * @param {?ResponseBodyInit=} body
+   * @param {!ResponseInit=} init
+   */
+  constructor(body, init = {}) {
+    const lowercasedHeaders = map();
+    const data = Object.assign({
+      status: 200,
+      statusText: 'OK',
+      responseText: (body ? String(body) : ''),
+      /**
+       * @param {string} name
+       * @return {string}
+       */
+      getResponseHeader(name) {
+        return hasOwn(lowercasedHeaders, name) ?
+          lowercasedHeaders[String(name).toLowerCase()] : null;
+      },
+    }, init);
+
+    init.status = init.status || 200;
+
+    if (isArray(init.headers)) {
+      init.headers.forEach(entry => {
+        const headerName = entry[0];
+        const headerValue = entry[1];
+        lowercasedHeaders[String(headerName).toLowerCase()] =
+            String(headerValue);
+      });
+    } else if (isObject(init.headers)) {
+      for (const key in init.headers) {
+        lowercasedHeaders[String(key).toLowerCase()] =
+            String(init.headers[key]);
+      }
+    }
+    if (init.status) {
+      data.status = parseInt(init.status, 10);
+    }
+    if (init.statusText) {
+      data.statusText = String(init.statusText);
+    }
+
+    super(/** @type {XMLHttpRequestDef} */(data));
+  }
+}
+
+/**
+ * Installs fetch and Response polyfill
+ *
+ * @export
+ * @param {Window} win
+ */
+export function install(win) {
+  if (!win.fetch) {
+    Object.defineProperty(win, 'fetch', {
+      value: fetch,
+      writable: true,
+      enumerable: false,
```

Comment:
`fetch` is [enumerable](https://github.com/ampproject/amphtml/pull/17468#discussion_r210111308). Also configurable.